client.go

```diff
@@ -50,10 +50,23 @@ type Pusher interface {
 	Push(ctx context.Context, f Function) (string, error)
 }
 
+type Status int
+
+const (
+	Failed Status = iota
+	Deployed
+	Updated
+)
+
+type DeploymentResult struct {
+	Status Status
```

Comment:
I also like this because it could helps us maintain backwards compatibility going forward